### PR TITLE
Draft PR: default-ts template for WXP on Toolkit

### PR DESCRIPTION
**Pull request by YueLi-MSFT**
Thank you for your pull request!  Please provide the following information.

---

**Change Description**:

    Describe what the PR is about. 

1. **Do these changes impact any *npm scripts* commands (in package.json)?** (e.g., running 'npm run start')
    If Yes, briefly describe what is impacted.

2. **Do these changes impact *VS Code debugging* options (launch.json)?**
    If Yes, briefly describe what is impacted.

3. **Do these changes impact *template output*?** (e.g., add/remove file, update file location, update file contents)
    If Yes, briefly describe what is impacted.

4. **Do these changes impact *documentation*?** (e.g., a tutorial on https://docs.microsoft.com/en-us/office/dev/add-ins/overview/office-add-ins)
    If Yes, briefly describe what is impacted.

If you answered yes to any of these please do the following:
    > Include 'Rick-Kirkham' in the review
    > Make sure the README file is correct

**Validation/testing performed**:

    Describe manual testing done.

### CODE REVIEW

File: .vscode/launch.json [hunk 1]
```diff
@@ -25,5 +25,68 @@
       "preLaunchTask": "Debug: Outlook Desktop",
       "postDebugTask": "Stop Debug"
     },
+    {
```
Comment (millerds) [inline .vscode/launch.json:28]: Should probably order the configs in the same ways as is done in the master branch to maintain uniformity.

File: package.json [hunk 2]
```diff
@@ -7,7 +7,6 @@
   },
   "license": "MIT",
   "config": {
-    "app_to_debug": "outlook",
```
Comment (millerds) [inline package.json:10]: This should not be removed for this special case.  If it doesn't look like it's needed then we should log a bug to remove it in general across all the templates.

**Review by millerds — CHANGES_REQUESTED** [review]
Note that the current json manifest branches were reduced from the master branch to target Outlook because that is all that was supported at the time.  When changing it back to support other host we should continue to follow the patterns used in general and if there is a desire to change the patterns then a task/bug should logged and discussed as a change for all templates.

We should be following the branch progression for changes . . . meaning changes for json manifest template goes to "json-preview" branch first.  Changes being made ready for public consumption should then be merged from "json-preview" to "json-preview-yo-office-prerelease" and verified using 'yo office' there before finally being pushed to "json-preview-yo-office" for customers to use.

On top of following the branch progression, we need to coordinate with the documentation team (Ricky) and managing the branches and Outlook template (which is going to GA in a couple of months) as they own the repo management now.

File: package.json [hunk 3]
```diff
@@ -25,7 +24,10 @@
     "signin": "office-addin-dev-settings m365-account login",
     "signout": "office-addin-dev-settings m365-account logout",
     "prettier": "office-addin-lint prettier",
-    "start": "office-addin-debugging start manifest.json",
+    "start:word": "office-addin-debugging start manifest.json --app word",
+    "start:excel": "office-addin-debugging start manifest.json --app excel",
+    "start:powerpoint": "office-addin-debugging start manifest.json --app powerpoint",
+    "start:outlook": "office-addin-debugging start manifest.json --app outlook",
```
[the remark this answers is not in the document]
Comment (QuanfuXiao) [reply package.json:30]: Thanks Darren, it is reasonable to leave one command in "package.json" and pass parameters in tasks config.

File: .vscode/launch.json (hunk 1 above)
Comment (QuanfuXiao) [reply .vscode/launch.json:28]: @millerds , should we use unique port for each config?

Comment (millerds) [reply .vscode/launch.json:28]: No . . . the ports listed are the standard cases for attaching to the launched webview host.  The chrome version and the legacy version each have that as their default.

I'm just asking that the order of each config match that of the master branch.

File: package.json (hunk 3 above)
[the remark this answers is not in the document]
Comment (millerds) [reply package.json:30]: That is what the master branch does . . . use that as your starting point.

File: .vscode/tasks.json [hunk 4]
```diff
@@ -34,10 +34,52 @@
         "showReuseMessage": false
       }
     },
+    {
```
[the remark this answers is not in the document]
Comment (YueLi-MSFT) [reply .vscode/tasks.json:37]: Thanks Millerds! Will do!

File: .vscode/tasks.json [hunk 5]
```diff
@@ -69,6 +111,20 @@
       },
       "problemMatcher": []
     },
+    {
+      "label": "Link",
+      "type": "shell",
+      "command": "npm link office-addin-dev-settings --force",
```
Comment (millerds) [inline .vscode/tasks.json:117]: I don't think we want to do this automatically as part running in VS Code.  This should be left for someone to use if they explicitly want to try a change to the scripts package locally.

Comment (QuanfuXiao) [reply .vscode/tasks.json:117]: Yeah, it is only for POC demo and will be deleted when TAOS sideload for WXP manifest is ready.

**Comment by QuanfuXiao** [comment]
Hi @millerds , regarding the branch issue, is it reasonable to add a new branch only for the toolkit? Since "json-preview-yo-office" should remain to be used in Yo Office for Outlook only.

**Comment by millerds** [comment]
> Hi @millerds , regarding the branch issue, is it reasonable to add a new branch only for the toolkit? Since "json-preview-yo-office" should remain to be used in Yo Office for Outlook only.

This is something that needs to be discussed with the docs team that jointly owns maintaining the templates.  I plan on figuring that out and we'll have to have a plan in place before modifying the existing branches.

File: .vscode/launch.json (hunk 1 above)
Comment (YueLi-MSFT) [reply .vscode/launch.json:28]: Got it. Thanks @millerds

File: .vscode/tasks.json [hunk 6]
```diff
@@ -60,8 +102,22 @@
     },
     {
       "label": "Install",
-      "type": "npm",
-      "script": "install",
+      "type": "shell",
```
Comment (millerds) [inline .vscode/tasks.json:105]: why is this needed.

File: manifest.json [hunk 7]
```diff
@@ -184,7 +184,7 @@
                     {
                       "id": "msgReadOpenPaneButton",
                       "type": "button",
-                      "label": "Show Taskpane",
+                      "label": "Hello Excel",
```
Comment (millerds) [inline manifest.json:187]: We should not be labeling this specific to one host.  This is used by all the hosts.

File: src/commands/commands.ts [hunk 8]
```diff
@@ -13,16 +13,17 @@ Office.onReady(() => {
...
-  const message: Office.NotificationMessageDetails = {
-    type: Office.MailboxEnums.ItemNotificationMessageType.InformationalMessage,
-    message: "Performed action.",
-    icon: "Icon.80x80",
-    persistent: true,
-  };
-
-  // Show a notification message
-  Office.context.mailbox.item.notificationMessages.replaceAsync("ActionPerformanceNotification", message);
+async function action(event: Office.AddinCommands.Event) {
```
Comment (millerds) [inline src/commands/commands.ts:16]: This template is used for Outlook too.  If you want to make a command sample for different hosts (something not done in current xml based templates) then you need to make the template handle all hosts, or make options for host specific stuff that is cleaned up by the convert script when instantiating template for a particular host.

File: .vscode/tasks.json (hunk 6 above)
Comment (QuanfuXiao) [reply .vscode/tasks.json:105]: Will change it back. We added a "npm link" script for demo, but it seems not work well with "npm install" ("npm link" will modify files in node-modules, but "file been used" issue often occur when executed after "npm install" task).  This change still cannot solve all the issue.

File: src/commands/commands.ts (hunk 8 above)
Comment (QuanfuXiao) [reply src/commands/commands.ts:16]: Yeah, we will add commands of all hosts later.

File: src/taskpane/taskpane.ts [hunk 9]
```diff
@@ -9,15 +9,78 @@ Office.onReady((info) => {
...
+  }
+  if (info.host === Office.HostType.Excel) {
+    document.getElementById("sideload-msg").style.display = "none";
+    document.getElementById("app-body").style.display = "flex";
+    document.getElementById("run").onclick = runExcel;
+  }
+  if (info.host === Office.HostType.PowerPoint) {
+    document.getElementById("sideload-msg").style.display = "none";
+    document.getElementById("app-body").style.display = "flex";
+    document.getElementById("run").onclick = runPowerPoint;
```
[the remark this answers is not in the document]
Comment (YueLi-MSFT) [reply src/taskpane/taskpane.ts:27]: Yes, we hope that the template can support WXPO apps for Teams Toolkit feature.

**Pull request closed without merging**